### PR TITLE
Add optional preference FORM_FIELDS

### PR DESCRIPTION
**Pull request by sestanley13**
## Type of change

Improvement for somebody else's extension. Extension author: @honzatmn

## Screencast

Screen Shot 2021-12-17 at 11 31 01 PM
Screen Shot 2021-12-17 at 11 32 22 PM
Screen Shot 2021-12-17 at 11 33 02 PM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Comment by fe9lix** [comment]
@honzatmn Would you mind also reviewing those changes?

File: extensions/productboard/src/index.tsx [hunk 1]
```diff
@@ -52,19 +52,54 @@ export default function Command() {
...
+    </Form>
+  );
+
+  const fields = FORM_FIELDS.split(/,\s*/g);
+
+  async function handleFormTemplateSubmit(values: Omit<Note, 'content'> & FormValues) {
+    const { title, tags, saidBy } = values;
+    let content = '';
+    fields.forEach((field, index) => {
+      content += `<b>${field}:</b> ${values[index.toString()]} <br />`;
```
Comment (honzatmn) [inline extensions/productboard/src/index.tsx:79]: Wouldn't it be better to format content here with markdown syntax, and let markdown converter in `handleSubmit` take care of the conversion to HTML? 

Also, have you tested markdown syntax in custom fields? Will it get converted properly?

File: extensions/productboard/src/index.tsx [hunk 2]
```diff
@@ -52,19 +52,54 @@ export default function Command() {
     }
   }
 
-  return (
+  const { FORM_FIELDS } = preferences;
+
+  if (!FORM_FIELDS) return (
```
Comment (honzatmn) [inline extensions/productboard/src/index.tsx:57]: There is a lot of repeated code with this approach. Would it be possible to clean it up a bit, so e.g. title, tags and saidBy fields are not defined twice?

**Review by honzatmn — CHANGES_REQUESTED** [review]
Hi Scott, thanks for your contribution, this is a great new feature! I added two small comments, otherwise it looks good.

**Comment by pernielsentikaer** [comment]
@sestanley13 could you look into the suggestion above?

**Comment by pernielsentikaer** [comment]
Hi @sestanley13 – are you able to look into the suggestions made by @honzatmn, so we can get this PR closed?

**Comment by mattisssa** [comment]
Hey @sestanley13! What is the status of this PR? It seems like it's been here for a long time unfinished. Shall we probably close this one and once you're ready – open another PR?

**Pull request closed without merging**